cluster-autoscaler/utils/gpu/gpu.go

```diff
@@ -47,47 +47,44 @@ const (
 	MetricsNoGPU = ""
 )
 
-// GetGpuTypeForMetrics returns name of the GPU used on the node or empty string if there's no GPU
+// GetGpuTypeForMetrics returns the name of the custom resource and the GPU used on the node or empty string if there's no GPU
 // if the GPU type is unknown, "generic" is returned
 // NOTE: current implementation is GKE/GCE-specific
-func GetGpuTypeForMetrics(GPULabel string, availableGPUTypes map[string]struct{}, node *apiv1.Node, nodeGroup cloudprovider.NodeGroup) string {
-	// we use the GKE label if there is one
-	gpuType, labelFound := node.Labels[GPULabel]
-	capacity, capacityFound := node.Status.Capacity[ResourceNvidiaGPU]
-
-	if !labelFound {
-		// no label, fallback to generic solution
-		if capacityFound && !capacity.IsZero() {
-			return MetricsGenericGPU
-		}
-
-		// no signs of GPU
-		return MetricsNoGPU
+func GetGpuTypeForMetrics(gpuConfig *cloudprovider.GpuConfig, availableGPUTypes map[string]struct{}, node *apiv1.Node, nodeGroup cloudprovider.NodeGroup) (string, string) {
+	// There is no sign of GPU
+	if gpuConfig == nil {
```

Comment:
nit: It took me a bit to figure out that the PR doesn't change behavior for the existing GPU logic (there's still 1 difference - this function only looks at capacity, while GetNodeGpuConfig utilizes NodeHasGpu which looks at allocatable - but capacity and allocatable should be in sync for GPUs, and allocatable is arguably more correct - so it looks fine to me). This function could really use a unit test, if you're up for it.

Reply:
This is a little bit tricky, started writing a test but I found out that previous PR #5459 introduced a "hidden" import cycle (so far the cycle happens only if you use the cloudprovider test package). If you're OK with it, I'd prefer to follow up in the next PR

Reply:
The cycle will probably have to be solved sooner or later, but a follow-up SGTM.